### PR TITLE
Feature/ELG-42/Implement package calculator for water heaters

### PR DESCRIPTION
**Pull request by FernandoRamirez-De-Aguilar**
- Updated journey steps

1. water heater category > sort question > Calculator or normal energy label form

- Implemented calculations

1. As per Matt's app.
2. Added QREF values for all `LoadProfile` from [eur-lex.europa.eu](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=celex%3A32013R0812)

- Tests

1. Tested the efficiency class for some upper/lower limits related to the `WaterHeatingEfficiencyPercentage`
2. Did some more tests for all values by placing random values on Matt's app and then asserting the same values on the Java app. Accounted for decimal precision.

SS:
image
image

### CODE REVIEW

**Review by matteason — CHANGES_REQUESTED** [review]
Couple of design changes on the form page but the calculations themselves look good, thanks very much @FernandoRamirez-De-Aguilar

File: src/main/java/uk/gov/beis/els/categories/waterheaters/model/WaterSolarPackagesCalculatorForm.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  @FieldPrompt("Declared load profile")
+  @NotBlank(message = "Select a declared load profile")
+  private String declaredLoadProfile;
+
+  @FieldPrompt("Can a hot water storage tank be included in this package?")
+  @NotNull(message = "Specify if a hot water storage tank can be included")
+  private Boolean storageTank;
+
+  @FieldPrompt("Can a solar collector be included in this package?")
```
[the remark this answers is not in the document]
Comment (FernandoRamirez-De-Aguilar) [reply src/main/java/uk/gov/beis/els/categories/waterheaters/model/WaterSolarPackagesCalculatorForm.java]: done

File: src/main/java/uk/gov/beis/els/categories/waterheaters/model/WaterSolarPackagesCalculatorForm.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import javax.validation.constraints.Digits;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import uk.gov.beis.els.categories.common.StandardTemplateForm50Char;
+import uk.gov.beis.els.model.meta.FieldPrompt;
+
+public class WaterSolarPackagesCalculatorForm extends StandardTemplateForm50Char {
+
+  @FieldPrompt("Water heating efficiency (%)")
```
[the remark this answers is not in the document]
Comment (FernandoRamirez-De-Aguilar) [reply src/main/java/uk/gov/beis/els/categories/waterheaters/model/WaterSolarPackagesCalculatorForm.java]: done

File: src/main/java/uk/gov/beis/els/categories/waterheaters/model/WaterSolarPackagesCalculatorForm.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  @Digits(integer = 3, fraction = 0, message = "Enter the water heating efficiency (%), up to 3 digits long")
+  private String waterHeatingEfficiencyPercentage;
+
+  @FieldPrompt("Declared load profile")
+  @NotBlank(message = "Select a declared load profile")
+  private String declaredLoadProfile;
+
+  @FieldPrompt("Can a hot water storage tank be included in this package?")
```
[the remark this answers is not in the document]
Comment (FernandoRamirez-De-Aguilar) [reply src/main/java/uk/gov/beis/els/categories/waterheaters/model/WaterSolarPackagesCalculatorForm.java]: done

File: src/main/resources/templates/categories/water-heaters/waterSolarPackagesCalculator.ftl [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    title="Packages of water heater and solar device energy label calculator"
+    showInsetText=false
+>
+    <@govukTextInput.textInput path="form.waterHeatingEfficiencyPercentage"/>
+
+    <@govukSelect.select path="form.declaredLoadProfile" options=loadProfile/>
+
+    <@govukRadios.radioYesNo path="form.storageTank"/>
+
+    <@govukTextInput.textInput path="form.annualNonSolarHeatContribution"/>
```
[the remark this answers is not in the document]
Comment (FernandoRamirez-De-Aguilar) [reply src/main/resources/templates/categories/water-heaters/waterSolarPackagesCalculator.ftl:15]: done

File: src/main/resources/templates/categories/water-heaters/waterSolarPackagesCalculator.ftl [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<#include '../../layout.ftl'>
+
+<@common.standardProductForm
+    title="Packages of water heater and solar device energy label calculator"
+    showInsetText=false
+>
+    <@govukTextInput.textInput path="form.waterHeatingEfficiencyPercentage"/>
```
[the remark this answers is not in the document]
Comment (FernandoRamirez-De-Aguilar) [reply src/main/resources/templates/categories/water-heaters/waterSolarPackagesCalculator.ftl:8]: done

File: src/main/resources/templates/categories/water-heaters/waterSolarPackagesCalculator.ftl [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    <@govukSelect.select path="form.declaredLoadProfile" options=loadProfile/>
+
+    <@govukRadios.radioYesNo path="form.storageTank"/>
+
+    <@govukTextInput.textInput path="form.annualNonSolarHeatContribution"/>
+
+    <@govukTextInput.textInput path="form.auxElectricityConsumption"/>
+</@common.standardProductForm>
+
+<#--<@defaultPage pageTitle="Do you need help calculating the energy rating for this package?">-->
```
[the remark this answers is not in the document]
Comment (FernandoRamirez-De-Aguilar) [reply src/main/resources/templates/categories/water-heaters/waterSolarPackagesCalculator.ftl]: how awkward

[the remark this answers is not in the document]
Comment (FernandoRamirez-De-Aguilar) [reply src/main/resources/templates/categories/water-heaters/waterSolarPackagesCalculator.ftl]: done

**Review by matteason — APPROVED** [review]

**Review by jamesbarnett91 — APPROVED** [review]

**Pull request merged**